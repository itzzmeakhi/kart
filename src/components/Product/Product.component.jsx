import React from 'react';
import { connect } from 'react-redux';

import { addToCart, removeFromCart } from './../../redux/cart/cart.actions';

import './Product.styles.scss';

const Product = ({ add, cartItems, remove, prodImg, prodName, prodPrice, prodId }) => {
    const productDetails = {
        prodId,
        prodName,
        prodImg,
        prodPrice
    }

    return (
        <div className = 'product'>
            <div className = 'product__img'>
                <img src = {prodImg} alt = {prodName} />
            </div>
            <div className = 'product__desc'>
                <div className = 'product__details'>
                    <p> {prodName} </p>
                    <p className = 'price'> Rs. {prodPrice} </p>
                </div>
                {cartItems.filter(itm => itm.prodId === prodId).length <= 0 ? (
                    <button
                        className = 'add__button'
                        onClick = {() => {
                            add(productDetails);
                        }}
                        type = 'button'>
                            Add
                    </button>
                ) : (
                    <button
                        className = 'remove__button'
                        onClick = {() => {
                            remove(prodId);
                        }}
                        type = 'button'>
                            Remove
                    </button>
                )}
            </div>
        </div>
    )
}

const mapStateToProps = state => {
    return {
        cartItems: state.cart.cartItems
    }
}

const mapDispatchToProps = dispatch => {
    return {
        add: (item) => dispatch(addToCart(item)),
        remove: (id) => dispatch(removeFromCart(id))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Product);