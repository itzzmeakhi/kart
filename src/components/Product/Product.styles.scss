@import './../../scss/colors';

.product {
    height: 380px;
    padding: 20px 10px 0px 10px;
    box-sizing: border-box;
    border: 1px solid $color-white-thick;
    border-radius: 5px;
    background-color: $color-white;
    margin: 10px;
    flex: 1 300px;
    display: flex;
    flex-direction: column;
    align-items: center;
    .product__img {
        height: 45%;
        width: 70%;
        img {
            max-width: 100%;
            max-height: 100%;
            display: block;
            margin: 0 auto;
        }
    }
    .product__desc {
        padding: 5px;
        margin-top: 10px;
        display: flex;
        box-sizing: border-box;
        color: $color-blue-thick;
        font-size: 90%;
        > * {
            margin: 0px 5px;
        }
        .price {
            background: $color-blue-thick;
            color: $color-white;
            padding: 2px;
            width: 60%;
            text-align: center;
        } 
        button {
            align-self: center;
            padding: 5px 20px;
            border-radius: 5px;
            color: $color-blue-thick;
            cursor: pointer;
            &:focus {
                outline: none;
            }
        }
        .add__button {
            background: transparent;
            border: 2px solid $color-blue-thick;
        }
        .remove__button {
            padding: 5px 10px;
            background: $color-red-shade;
            color: $color-white;
            border: none;
        }
    }
}