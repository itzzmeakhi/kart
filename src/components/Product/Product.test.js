import React from 'react';
import { shallow } from 'enzyme';

import Product from './Product.component';
import { storeFactory } from './../../test.utils';


const setup = (initialState={}) => {
    const store = storeFactory(initialState);
    const props = {
        prodId: 'abc',
        prodName: 'prod1',
        prodImg: 'https://www.google.com',
        prodPrice: 299
    }
    const wrapper = shallow(<Product store = {store} {...props} />).dive();
    return wrapper;
}


describe('<Product />', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = setup();
    })

    it('should component rendered without errors', () => {
        expect(wrapper.length).toBe(1);
    })

    it('should have product image', () => {
        expect(wrapper.dive().find('.product__img').length).toBe(1);
    })

    it('should have product description', () => {
        expect(wrapper.dive().find('.product__desc').length).toBe(1);
    })

    it('should have valid props passed', () => {
        expect(wrapper.props().prodId).toEqual('abc');
        expect(wrapper.props().prodName).toEqual('prod1');
        expect(wrapper.props().prodImg).toEqual('https://www.google.com');
        expect(wrapper.props().prodPrice).toEqual(299);
    })    
    
    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

})