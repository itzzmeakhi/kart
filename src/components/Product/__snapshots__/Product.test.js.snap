// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> should match the snapshot 1`] = `ShallowWrapper {}`;
